src/Kestrel.Core/Internal/Http/Http1OutputProducer.cs

```diff
@@ -89,7 +89,7 @@ public void Write<T>(Func<PipeWriter, T, long> callback, T state)
             }
         }
 
-        public Task WriteAsync<T>(Func<PipeWriter, T, long> callback, T state)
+        public Task WriteAsync<T>(Func<PipeWriter, T, long> callback, T state, CancellationToken cancellationToken)
```

Comment:
This is only used for headers and therefore isn't used.

Reply:
Why do you say that? Theoretically, you can change the response buffer to a small number and cause WriteAsync to hang on FlushAsync here https://github.com/aspnet/KestrelHttpServer/blob/80c8e83c689a8e2ed52e1c4b7b84c4c2d8ad15ee/src/Kestrel.Core/Internal/Http/Http1OutputProducer.cs#L107

Reply:
My bad, for some reason I thought this was (only) called by WriteResponseHeaders which would never use a CancellationToken. Forgot this was used to write chunked response bodies.

Looking at the code now, I realize we should also just remove the `Write<T>` overload. That's what we used to use for header writing prior to the HTTP/2 refactoring.

It looks like we regressed CancellationToken handling for chunked response bodies in 2.1.0. Maybe we should backport this fix.